### PR TITLE
Feat/slider new

### PR DESCRIPTION
**Pull request by qradle-yndx**

### CODE REVIEW

File: src/blocks/SliderNew/Slider.scss [hunk 1]
```diff
@@ -0,0 +1,278 @@
+@import '../../../styles/mixins';
+@import '../../../styles/variables';
+
+$slideOffset: 8px;
```
Comment (NikitaCG) [inline src/blocks/SliderNew/Slider.scss:4]: we are use this variable in one place, I think we can use in place and use `$indent[...]` variable

File: src/blocks/SliderNew/Slider.tsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        blockClassName,
+        arrowSize,
+        onSlideChange,
+        onSlideChangeTransitionStart,
+        onSlideChangeTransitionEnd,
+        onActiveIndexChange,
+        onBreakpoint,
+    } = props;
+
+    const [slider, setSlider] = useState<SwiperClass>();
```
Comment (NikitaCG) [inline src/blocks/SliderNew/Slider.tsx:66]: Let's make a custom hook https://github.com/gravity-ui/page-constructor/pull/959/files#diff-785d9dd01e6dbd44a2c8a6d7437b73be9705f7b86a7de2624670a97a88113edaR66-R89

File: src/blocks/SliderNew/utils.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        xs: !mobileFullscreen && contentLength > 1 ? DEFAULT_SLIDE_BREAKPOINTS.xs : 1,
+    };
+
+    return Object.entries(showCount).reduce((res, [breakpointName, value]) => {
+        // eslint-disable-next-line no-param-reassign
+        res[BREAKPOINTS[breakpointName as SliderBreakpointNames] + 1] = {
+            slidesPerView: value,
+        };
+        return res;
+    }, {} as Record<number, {slidesPerView: number}>);
```
Comment (NikitaCG) [inline src/blocks/SliderNew/utils.ts:47]: maybe there is a type for this in the Swiper repo?

File: src/blocks/SliderNew/Slider.tsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        Refable<HTMLDivElement>,
+        ClassNameProps {
+    type?: string;
+    anchorId?: string;
+    dotsClassName?: string;
+    blockClassName?: string;
+    arrowSize?: number;
+}
+
+export const SliderNewBlock = (props: PropsWithChildren<SliderNewProps>) => {
```
Comment (NikitaCG) [inline src/blocks/SliderNew/Slider.tsx:43]: up to you

we can do it right away like this
```
= ({
        animated,
        title,
        description,
        type,
        anchorId,
        arrows = true,
        adaptive,
        autoplay = undefined,
        dots = true,
        dotsClassName,
        disclaimer,
        children,
        blockClassName,
        arrowSize,
        onSlideChange,
        onSlideChangeTransitionStart,
        onSlideChangeTransitionEnd,
        onActiveIndexChange,
        onBreakpoint,
    }: PropsWithChildren<SliderNewProps>) =>
    
```

File: src/blocks/SliderNew/Slider.tsx [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+                blockClassName,
+            )}
+        >
+            {anchorId && <Anchor id={anchorId} />}
+            <Title
+                title={title}
+                subtitle={description}
+                className={b('header', {'no-description': !description})}
+            />
+            <AnimateBlock className={b('animate-slides')} animate={animated}>
```
Comment (NikitaCG) [inline src/blocks/SliderNew/Slider.tsx:109]: pagination dots over slide animation

you need some kind of solution, either show them after the end of the animation or also animate, try the options

**Review by NikitaCG — CHANGES_REQUESTED** [review]

File: src/blocks/SliderNew/schema.ts [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+} from '../../schema/validators/common';
+
+const LoadableProps = {
+    additionalProperties: false,
+    required: ['source'],
+    properties: {
+        source: {
+            type: 'string',
+            // add loadable sources here if you use it in your project
+            // enum: ['my-loadable-source-1', 'my-loadable-source-1'],
```
Comment (aeksandla) [inline src/blocks/SliderNew/schema.ts:17]: Is this really necessary here?

File: src/blocks/SliderNew/schema.ts [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        params: {
+            type: 'object',
+            patternProperties: {
+                '.*': {
+                    type: ['string', 'number', 'boolean'],
+                },
+            },
+        },
+    },
+    // remove it in your custom validator schema if you use loadable
```
Comment (aeksandla) [inline src/blocks/SliderNew/schema.ts:34]: Is this really necessary here?

File: src/blocks/SliderNew/schema.ts (hunk 6 above)
Comment (qradle-yndx) [reply src/blocks/SliderNew/schema.ts:17]: Let's keep these comments (this one and the one below) as long as slider-new retains this logic from old slider

**Review by aeksandla — APPROVED** [review]

File: src/blocks/SliderNew/Slider.tsx (hunk 5 above)
Comment (qradle) [reply src/blocks/SliderNew/Slider.tsx:122]: Dots animation added

File: src/blocks/SliderNew/Slider.tsx [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+    onBreakpoint,
+}: PropsWithChildren<SliderNewProps>) => {
+    const {childrenCount, breakpoints, slider, autoplay, onSwiper} = useSlider({
+        slidesToShow,
+        children,
+        type,
+        autoplayMs,
+    });
+
+    const [isLocked, setIsLocked] = useState(false);
```
[the remark this answers is not in the document]
Comment (qradle) [reply src/blocks/SliderNew/Slider.tsx]: also moved handleNext, handlePrev to hook

**Review by NikitaCG — APPROVED** [review]

**Pull request merged**